lib/include/platform/atomic.h

```diff
@@ -0,0 +1,299 @@
+/*
```

Comment:
Consider moving this file to platform/, as it seems specific to gcc.

Reply:
```platform/include/ ``` could be a potential candidate. This folder only has ```posix``` dir, where atomic doesn't fit. Thoughts on how to organize them? Or ```platform/include/atomic.h``` is good enough for now?